goblin-breakdown/readme.md

```diff
@@ -0,0 +1,80 @@
+# Goblin Breakdown
+
+Use AI to break down coding problems into smaller pieces.
+
+<!--{{<note type="tip" title="Prompt">}}>-->
+
+If Goblin is offline, use any LLM with the pre-prompt "Break down this task into single responsibility functions".
+
+<!--{{</note>}}-->
+
+## Learning Objectives
+
+```objectives
+- [ ] Use AI to break down a coding problem
+- [ ] Write comments to structure small single responsibility functions
+- [ ] Produce a planned solution to a coding problem
+```
+
+## Requirements
+
+We're practising breaking down our problems again. It must be Saturday!
+
+This time we're going to use AI to help us. [Goblin](https://goblin.tools) is a small, clever integration with ChatGPT that makes magic todo lists. We're going to use it to break down our problem and plan our solution.
+
+### 🧰 1. Setup
+
+You should already be set up but just in case, you need to open this repository in VS Code.
+
+<!--{{<note type="activity" title="Getting Set Up">}}>-->
+
+1.  Set a whole class [timer for 5 minutes](https://www.google.com/search?q=timer+for+5+minutes).
+1.  Fork this repository to your own GitHub account if you have not already.
+1.  Clone the repository to your machine.
+1.  Open in VS Code and go to the `goblin-breakdown` folder.
+
+<!--{{</note>}}-->
+
+## 👣 2. Break down coding problems
+
+Let's try breaking down a coding problem. Let's do a sample problem together. Set a whole [class timer for 10 minutes](https://www.google.com/search?q=15+minute+timer).
+
+<!--{{<note type="activity" title="Planning a Solution 10m">}}>-->
+
+### 🧱 3. Plan a solution
+
+> Format a UK address and UK phone number from a JSON object and show it as HTML on the page
+
+1. Paste the above problem into Goblin and press the 🪄 wand button.
+1. Goblin will give you some steps. In the `contact` folder, paste each of the steps into `index.js` as comments. There are some samples given but your Goblin might be a bit different.
+1. You likely have around five steps. Under each comment, write the name of the function you will write to achieve the step. Two are given as an example. Don't write the function out, just the name and leave the function empty.
+1. Now you have planned your code! _If you have time_, start writing the functions.
+
+<!--{{</note>}}-->
+
+<!--{{<note type="tip" title="Sample Solution">}}>-->
+
+There isn't a "right answer" to this workshop, but if you want to see a basic solution, look in the `contact` folder for the file `completed.js`.
+
+<!--{{</note>}}-->
+
+### 🧠 4. How can you use this?
+
+Set a whole [class timer for 10 minutes](https://www.google.com/search?q=15+minute+timer).
+
+<!--{{<note type="activity" title="Reflect and Discuss 10m">}}>-->
+
+Reflect on this process of breaking down the problem and planning your solution. Discuss with your pair or group.
```

Comment:
One more reflection I'm not sure how to prompt - I'm generally of the view that this kind of breaking down _gives you easier sub-problems to Google / ask ChatGPT_.

"How do I set the text of a created HTML element?" - easy to Google
"How do I show an address in a page?" - harder to Google

I'm not sure there's an easy way to prompt this reflection without having the trainees actually do the implementation, but I'd be curious as to your thoughts...

Reply:
Yeah, I agree it's much easier. We can pick that up in another workshop.